src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDateArray = [];
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = dateArray[fromFormat.indexOf('YY')];
+
+    dateArray[fromFormat.indexOf('YY')] = year < 30 ? '20' + year : '19' + year;
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    const year = dateArray[fromFormat.indexOf('YYYY')];
+
+    dateArray[fromFormat.indexOf('YYYY')] = year[2] + year[3];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    for (let j = 0; j < toFormat.length - 1; j++) {
+      if (
+        fromFormat[i] === toFormat[j] ||
+        (fromFormat[i] === 'YY' && toFormat[j] === 'YYYY') ||
+        (fromFormat[i] === 'YYYY' && toFormat[j] === 'YY')
+      ) {
+        newDateArray[j] = dateArray[i];
+      }
+    }
+  }
+
+  return newDateArray.join(toFormat[3]);
```

Comment:
The formatDate function does not handle cases where the `fromFormat` and `toFormat` arrays have different orders of date parts. The function should reorder the date parts according to the `toFormat` array. Additionally, there is no check to ensure that the `fromFormat` and `toFormat` arrays have valid and matching date parts before attempting to reformat the date. This can lead to incorrect results or runtime errors if the formats are not compatible.